packages/validation-manager/src/TracerResultParser.ts

```diff
@@ -220,23 +239,22 @@ export function tracerResultParser (
       ValidationErrors.OpcodeValidation)
   }
 
-
   const sender = userOp.sender.toLowerCase()
   // stake info per "number" level (factory, sender, paymaster)
   // we only use stake info if we notice a memory reference that require stake
   const stakeInfoEntities = {
-    factory: validationResult.factoryInfo,
-    account: validationResult.senderInfo,
-    paymaster: validationResult.paymasterInfo
+    [userOp.factory!]: validationResult.factoryInfo,
```

Comment:
This code implicitly assume entities are unique (paymaster.address != factory.address), otherwise its behaviour is undefined.
I think it is a good thing, but needs an explicit check (when statically validating the tx, and probably define it in the spec too)

Reply:
Yes, this is assumed. I checked and it does not seem like this specific place will have a problem, as the struct is mostly used to check if the address is staked. 
I did create a separate issue to make sure we agree on what to do in this situation:
https://opengsn.atlassian.net/browse/AA-386